src/components/stegvisning/stegvisning-hooks.tsx

```diff
@@ -24,8 +34,61 @@ export const useStegvisningNavigation = (currentPath: Path) => {
         : stegvisningOrder
     navigate(stepArrays[stepArrays.indexOf(currentPath) + 1])
   }
+
   const onStegvisningPrevious = () => {
-    navigate(-1)
+    let antallStepTilbake = 1
+    const isEndring = loependeVedtak && isLoependeVedtakEndring(loependeVedtak)
+    const stepArrays = isEndring ? stegvisningOrderEndring : stegvisningOrder
+
+    const currentPathIndex = stepArrays.indexOf(currentPath)
+
+    // Hvis brukeren er på eller forbi afp steget
+    if (
+      (!isEndring && currentPathIndex >= 4) ||
+      (isEndring && currentPathIndex >= 2)
+    ) {
+      // Bruker med uføretrygd som er eldre enn AFP-Uføre oppsigelsesalder har ikke fått steg om AFP og skal navigere tilbake forbi den
+      if (
+        ufoeregrad &&
+        foedselsdato &&
+        isFoedselsdatoOverEllerLikMinUttaksalder(foedselsdato)
+      ) {
+        antallStepTilbake = antallStepTilbake + 1
+      } else if (loependeVedtak?.afpPrivat || loependeVedtak?.afpOffentlig) {
+        // Bruker med vedtak om AFP har ikke fått steg om AFP og skal navigere tilbake forbi den
+        antallStepTilbake = antallStepTilbake + 1
+      }
+    }
+
+    // Hvis brukeren er på eller forbi ufoeretrygdAFP steget
+    if (
+      (!isEndring && currentPathIndex >= 5) ||
+      (isEndring && currentPathIndex >= 3)
+    ) {
+      // Bruker uten uføretryg, eller bruker med uføretrygd som har svart "nei" på AFP, eller bruker med uføretrygd eldre enn AFP-Uføre oppsigelsesalder har ikke fått infosteg om AFP + Uføretrygd og skal navigere tilbake forbi den
```

Comment:
Siden det er vanlig å ikke bruke line wrapping, hadde det vært fint med et linjeskift i denne kommentaren 😄
```suggestion
      // Bruker uten uføretrygd, eller bruker med uføretrygd som har svart "nei" på AFP, eller bruker med uføretrygd
      // eldre enn AFP-Uføre oppsigelsesalder har ikke fått infosteg om AFP + Uføretrygd og skal navigere tilbake forbi den
```